semantic-conventions/src/opentelemetry/semconv/model/utils.py

```diff
@@ -48,3 +48,34 @@ def check_no_missing_keys(yaml, mandatory):
         position = yaml.lc.data[list(yaml)[0]]
         msg = "Missing keys: {}".format(missing)
         raise ValidationError.from_yaml_pos(position, msg)
+
+
+class ValidatableYamlNode:
```

Comment:
This is the base class for SemanticConvention (and also UnitMember). In the future, I expect it to be the base class for MetricLabel. 
It owns validation of keys (that all the mandatory keys are present in a yaml node, and that only allowed keys are present), and the validation of values (that the values of the constructed object are allowed).

It may be worth making SemanticAttribute derive from this class, but I didn't do that work now.